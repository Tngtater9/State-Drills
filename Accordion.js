import React from 'react';
import './Accordion.css'

class Accordion extends React.Component {
    static defaultProps = {sections : []}
    state = {currentSectionIndex: null}
    handleClick (index) {
        this.setState({currentSectionIndex: index})
    }
    renderSection () {
        return this.props.sections.map((section, index) => (
            <li>
            <button key={index} onClick = {() => this.handleClick(index)}>{section.title}</button>
            {index === this.state.currentSectionIndex ? this.renderContent() : ""}
            </li>
            )
        )
    }
    renderContent () {
        const currentSection = this.props.sections[this.state.currentSectionIndex]
        return (
            <p>
                {currentSection.content}
            </p>
        )
    }
    render () {
        return (
            <ul>
                {this.renderSection()}
            </ul>
        )
    }
}

export default Accordion