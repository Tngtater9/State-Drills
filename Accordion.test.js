import React from 'react';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Accordion Component',() => {
    const sections = [
        {
        title: 'The title',
        content: 'The content is here'
      },
      {
        title: 'The second title',
        content: 'The content is here too'
      },
      {
        title: 'The third title',
        content: 'The content is here also'
      }
      ]
    it('renders an empty li when the sections prop is not supplied', () => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('renders no sections as active by default', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('opens a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('only opens the last section when multiple sections have been clicked', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
