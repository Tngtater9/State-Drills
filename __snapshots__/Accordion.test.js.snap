// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      The title
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      The second title
    </button>
    <p>
      The content is here too
    </p>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      The third title
    </button>
  </li>
</ul>
`;

exports[`Accordion Component opens a clicked section 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      The title
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      The second title
    </button>
    <p>
      The content is here too
    </p>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      The third title
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders an empty li when the sections prop is not supplied 1`] = `<ul />`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      The title
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      The second title
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      The third title
    </button>
  </li>
</ul>
`;
